src/coreclr/vm/multicorejit.cpp

```diff
@@ -1434,7 +1435,24 @@ void MulticoreJitManager::RecordMethodJit(MethodDesc * pMethod)
 
     if (m_pMulticoreJitRecorder != NULL)
     {
-        m_pMulticoreJitRecorder->RecordMethodJit(pMethod, true);
+        m_pMulticoreJitRecorder->RecordMethodJit(pMethod, true, true);
+
+        if (m_pMulticoreJitRecorder->IsAtFullCapacity())
+        {
+            m_fRecorderActive = false;
+        }
+    }
+}
+
+void MulticoreJitManager::RecordMethodLoad(MethodDesc * pMethod)
+{
+    STANDARD_VM_CONTRACT;
+
+    CrstHolder hold(& m_playerLock);
+
+    if (m_pMulticoreJitRecorder != NULL)
+    {
+        m_pMulticoreJitRecorder->RecordMethodJit(pMethod, true, false);
```

Comment:
I think the `dojit` parameter should be `true` here, in the sense we are recording a load of pregenerated method code, but the profile may be played back in a situation where that method is not pregenerated. The closest alternative would be to JIT the method, in effect the profile only records methods whose code should be loaded or jitted. Also, if the player does not try to compile the method, it would not actually load the pregenerated code, so it would only load the types relevant to where the method is, and would not load any other types used by the method. It may also help to parallelize the R2R method code loads.

I suggest the following:
- Remove the `dojit` parameters and treat it as true
- Remove the `NO_JIT_TAG` flag
- Remove `RecordMethodLoad` since the `dojit` parameter is the only differentiating factor
- Rename `RecordMethodJit` and equivalent to `RecordMethodLoadOrJit` or similar